### PR TITLE
Fix CodeQL scan error

### PR DESCRIPTION
**Pull request by liamtoozer**
### What is the context of this PR?
Our CodeQL is currently failing. We need to update the workflow so that we are using the latest version. Also, I have added a badge for visibility

### How to review
Check that the [CodeQL passes](https://github.com/ONSdigital/eq-submission-confirmation-consumer/actions/workflows/codeql-analysis.yaml)

### CODE REVIEW

File: .github/workflows/codeql-analysis.yaml [hunk 1]
```diff
@@ -13,10 +13,10 @@ name: "CodeQL"
 
 on:
   push:
-    branches: [main]
+    branches: [main, fix-codeql-scan-error]
```
Comment (liamtoozer) [inline .github/workflows/codeql-analysis.yaml]:
```suggestion
    branches: [main]
```
Reminder: Revert before merge

File: .github/workflows/codeql-analysis.yaml [hunk 2]
```diff
@@ -13,10 +13,10 @@ name: "CodeQL"
...
   pull_request:
     # The branches below must be a subset of the branches above
-    branches: [main]
+    branches: [main, fix-codeql-scan-error]
```
Comment (liamtoozer) [inline .github/workflows/codeql-analysis.yaml]:
```suggestion
    branches: [main]
```
Reminder: Revert before merge

File: .github/workflows/codeql-analysis.yaml [hunk 3]
```diff
@@ -13,55 +13,81 @@ name: "CodeQL"
...
-      #- run: |
-      #   make bootstrap
-      #   make release
+    # If the analyze step fails for one of the languages you are analyzing with
+    # "We were unable to automatically build your code", modify the matrix above
+    # to set the build mode to "manual" for that language. Then modify this step
+    # to build your code.
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun
+    - if: matrix.build-mode == 'manual'
```
Comment (berroar) [inline .github/workflows/codeql-analysis.yaml]: Do we need this step if we're not using manual?

Comment (liamtoozer) [reply .github/workflows/codeql-analysis.yaml]: I initially updated this file with the (mostly) bare minimum config needed but I noticed that the original file had boilerplate config, so I kept those sorts of parts in again here. I could remove the non-necessary config parts like this if we don't need them though?

Comment (berroar) [reply .github/workflows/codeql-analysis.yaml]: Yeah I see what you mean, though I don't think the previous one had any blocks that wouldn't ever get run so I'd probably remove it

Comment (liamtoozer) [reply .github/workflows/codeql-analysis.yaml]: Ah okay cool, will update it now, thanks 👍

**Review by berroar — APPROVED** [review]
LGTM 👍

**Review by Yuyuutsu — APPROVED** [review]

**Review by VirajP1002 — APPROVED** [review]

**Pull request merged**